db/migrate/20140108010302_add_token_to_projects.rb

```diff
@@ -2,7 +2,7 @@ class AddTokenToProjects < ActiveRecord::Migration
   def change
     change_table :projects do |t|
       t.string :token
-      t.index :token
+      t.index :token, length: 191
```

Comment:
we have done full indices before ... just need to set

mysql -u root -e "set GLOBAL innodb_large_prefix = true"
mysql -u root -e "set GLOBAL innodb_file_per_table = true"
mysql -u root -e "set GLOBAL innodb_file_format = 'barracuda'"

as we do in all other .travis.yml




On Tue, Jun 2, 2015 at 9:27 PM, Roman Sandler <notifications@github.com>
wrote:

> In db/migrate/20140108010302_add_token_to_projects.rb
> <https://github.com/zendesk/samson/pull/434#discussion_r31592996>:
>
> > @@ -2,7 +2,7 @@ class AddTokenToProjects < ActiveRecord::Migration
> >    def change
> >      change_table :projects do |t|
> >        t.string :token
> > -      t.index :token
> > +      t.index :token, length: 191
>
> indices are limited to 767 bytes in mysql and each character is 4 bytes,
> 767/4 = 191.75. Unfortunately Rails just defaults to 255.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/434/files#r31592996>.
>